packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/useChartCartesianAxisRendering.selectors.ts

```diff
@@ -54,6 +55,14 @@ const selectorChartYZoomOptionsLookup = createSelector(
 export const selectorChartZoomOptionsLookup = createSelector(
   [selectorChartXZoomOptionsLookup, selectorChartYZoomOptionsLookup],
   (xLookup, yLookup) => ({ ...xLookup, ...yLookup }),
+  {
+    memoizeOptions: {
+      // This selector returns Record<AxisId, DefaultizedZoomOptions> which is a map.
+      // Whenever the component re-renders, the axis is often a new object, which makes this a new instance.
+      // We need to use a custom equality check to avoid re-rendering the chart.
+      resultEqualityCheck: isDeepEqual,
```

Comment:
I think we might be able to achieve the same solution with a different approach. 

We're usually getting the `zoomOptions` only to later access a specific axis' options by doing `zoomOptions[axis.id]`. Would it make sense to create a selector for that directly (e.g., `selectorChartAxisZoomOptions`)? 

That way in most cases we would skip creating this object and the functions would receive the zoom options for the given axis. Plus, we would also avoid using deep equal here.

Reply:
I would rather not change it now as I use it in a different way on another PR, which this PR was extracted from.

But I like the idea and will try to test it there